ArtOfIllusion/src/artofillusion/procedural/ProceduralTextureCommon.java

```diff
@@ -0,0 +1,151 @@
+/* Copyright (C) 2017 by Maksim Khramov
+
+   This program is free software; you can redistribute it and/or modify it under the
+   terms of the GNU General Public License as published by the Free Software
+   Foundation; either version 2 of the License, or (at your option) any later version.
+   This program is distributed in the hope that it will be useful, but WITHOUT ANY 
+   WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A 
+   PARTICULAR PURPOSE.  See the GNU General Public License for more details. */
+
+package artofillusion.procedural;
+
+import artofillusion.MaterialPreviewer;
+import artofillusion.TextureParameter;
+import artofillusion.image.ImageMap;
+import artofillusion.math.RGBColor;
+import artofillusion.texture.Texture;
+import static artofillusion.texture.Texture.*;
+import artofillusion.ui.ActionProcessor;
+import artofillusion.ui.Translate;
+import artofillusion.ui.ValueSelector;
+import buoy.event.ValueChangedEvent;
+import buoy.widget.BDialog;
+import buoy.widget.BorderContainer;
+import buoy.widget.LayoutInfo;
+import buoy.widget.RowContainer;
+import java.awt.Rectangle;
+
+/**
+ *
+ * @author MaksK
+ */
+public class ProceduralTextureCommon
+{
+  private ProceduralTextureCommon()
+  {    
+  }
+  
+  /**
+   * Create a Procedure object for texture.
+   */
+
+  public static Procedure createTextureProcedure()
```

Comment:
It does make sense to associate this code with the texturing system, rather than in the procedure system. I'm just not sure that reducing the amount of duplicated code is enough reason in this instance to have an entirely separate helper class.